rtsp.go

```diff
@@ -658,11 +678,18 @@ func newRTSPCamera(ctx context.Context, _ resource.Dependencies, conf resource.C
 	}
 	cancelCtx, cancel := context.WithCancel(context.Background())
 	reader := gostream.VideoReaderFunc(func(_ context.Context) (image.Image, func(), error) {
-		latest := rc.latestFrame.Load()
+		latest := rc.latestOutput.Load()
```

Comment:
If we are calling `latestOutput.Store()` faster than we are calling `latestOutput.Load()` will we leak memory or do excessive garbage collection?

Reply:
No memory leak, but maybe GC

I think this would happen if something like:

Say I had a live stream going at 30fps, then I changed the frame rate in the Gost admin panel to 15fps. The equilibrium rate/number of items in the pool would be about halved then. So if half of those frames aren't being used anymore, they will be garbage collected properly during runtime by the finalizer. Whether that's excessive or not I am not entirely sure.